pkg/scheduler/partition.go

```diff
@@ -608,7 +608,7 @@ func (pc *PartitionContext) addNodeResources(node *objects.Node) {
 	if pc.totalPartitionResource == nil {
 		pc.totalPartitionResource = node.GetCapacity().Clone()
 	} else {
-		pc.totalPartitionResource.AddTo(node.GetCapacity())
```

Comment:
Should we replace `AddTo` by `resources.Add` for all? After that, we can remove `AddTo`

Reply:
`AddTo()` & its pairs have their use, see `total` in `removeAllocation()`. That's a local resource object which will not be stored anywhere and its lifecycle ends after the method returns. So this is how it's supposed to be used. But for variables which are potentially read&processed by other goroutines, using the immutable approach is safer because it just works without `Clone()`.

Reply:
`removeAllocation` it seems to me `total` can be initialized to `nil`, and then we just assign the new value (`delta`/`alloc.GetAllocatedResource()`) to `total`. Hence, `AddTo` can get removed from `removeAllocation`

Reply:
I think in this case, the mutation is happening under lock within updatePartitionResource(), which is fine. Instead, we should add Clone() to GetTotalPartitionResource(). We should never return a mutable reference from that.